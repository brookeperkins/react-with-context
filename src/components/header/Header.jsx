/* eslint-disable max-len */
import React from 'react';
import { useTheme } from '../../hooks/themes';
import theme from '../Theme/ThemeContainer.css';

const Header = () => {
  const { toggle } = useTheme();

  return (
    <header className={theme.dark}>
      <button data-testid="button" onClick={toggle}>Toggle Themes</button>
    </header>
  );
};
export default Header; 
