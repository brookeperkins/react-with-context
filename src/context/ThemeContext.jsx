import React from 'react';
import { useState } from 'react';

export const ThemeContext = React.createContext();

// eslint-disable-next-line react/prop-types
export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');

  const toggle = () => {
    if(theme === 'light') setTheme('dark');
    if(theme === 'dark') setTheme('light');
  };

  const bucket = {
    theme,
    toggle
  };

  return (
    <ThemeContext.Provider value={bucket}>
      {children}
    </ThemeContext.Provider>
  );
}; 
