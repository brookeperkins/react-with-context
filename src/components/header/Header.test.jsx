import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';
import { ThemeProvider } from '../../context/ThemeContext';

describe('<Header />', () => {
  it('renders the Header', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Header />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 
