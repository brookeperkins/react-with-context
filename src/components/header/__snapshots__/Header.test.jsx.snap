// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the Header 1`] = `
<DocumentFragment>
  <h1>
    Hey Arnold!
  </h1>
  ;
</DocumentFragment>
`;
